src/hooks/useSolanaBalance.ts

```diff
@@ -0,0 +1,47 @@
+import { PublicKey } from '@solana/web3.js';
+import { useQuery, UseQueryOptions } from '@tanstack/react-query';
+
+import { useSolanaConnection } from '@/hooks/useSolanaConnection';
+
+type BalanceProps = {
+  address: string | undefined;
+  token: string | undefined;
+  query?: Pick<UseQueryOptions, 'enabled'>;
+};
+
+export const useSolanaTokenBalance = ({ address, token }: BalanceProps) => {
+  const connection = useSolanaConnection();
+
+  const queryFn = async () => {
+    try {
+      if (!address || !token) {
+        throw new Error('Account or token address is not present');
+      }
+      const accountOwner = new PublicKey(address);
+      const tokenMint = new PublicKey(token);
+      const tokenAccounts = await connection.getParsedTokenAccountsByOwner(accountOwner, {
+        mint: tokenMint,
+      });
+      return {
+        data: {
+          formatted: tokenAccounts.value[0].account.data.parsed.info.tokenAmount.uiAmount,
```

Comment:
[q] can we get a description/link to the docs where this is described? i'm curious as to when we'd want to use a nonzero index here

Reply:
There's a bug here, I'll address this fix the issue and link a description to some docs tomorrow. The vast majority who have used USDC will only have one token account, and those who have not will have none (the bug).

Reply:
@yogurtandjam I've just updated this it should be fixed and more robust now. 

The vast majority will have 0 or 1 token accounts for a specific mint. In the unlikely case where a user has more than one token account, we choose the token account that has the highest balance. 

It's pretty rare that a user will have more than one; this quirk was caused in the early days on Solana when dapps created accounts without checking if users had one. Now they mostly do, and it costs rent to do so, there's a financial incentive to reuse existing accounts..

Here are some docs about how [tokens work on Solana](https://solana.com/docs/core/tokens).

Reply:
interesting, thanks for the background and a link to the docs! I know I could be more proactive in learning about the solana ecosystem so I appreciate your help on that front.